avocado/core/remote/runner.py

```diff
@@ -325,8 +329,8 @@ class VMTestRunner(RemoteTestRunner):
     Test runner to run tests using libvirt domain
     """
 
-    def __init__(self, job, result_proxy):
-        super(VMTestRunner, self).__init__(job, result_proxy)
+    def __init__(self, job, result_proxy, result):
```

Comment:
I agree that it's only a part of the complete switch.  I thought it made sense up to this point, because, as can be seen on `avocado/plugins/run.py`, we drop the hack to get one of the output plugins out of the result proxy, and instead, use the only one result instance available in the job results.

Also, I thought this change was necessary because it kept the code more honest.  Let me explain it better: because or plugin order initialization, we'd usually get the `HumanResult` instance out of the result proxy, and pass it to the other plugins, such as the `HTML` plugin.

The problem is, the `HumanResult` had some attributes populated (because of initialization times) that a plain `Result` wouldn't.  The HTML plugin, for instance, would crash looking for a `.logfile` attribute in a plain `Result` instance.  So it helped to reveal some bugs.

Finally, I think the difference is that, with this approach, we can be 100% sure that a plain `Result` instance has what the (new style) result plugins need.